addons/lunch/report/lunch_product_report.py

```diff
@@ -81,13 +81,12 @@ def init(self):
                     users.id AS user_id,
                     fav.user_id IS NOT NULL AS is_favorite,
                     product.new_until >= current_date AS is_new,
-                    max(l_order.date) AS last_order_date
+                    orders.last_order_date
                 FROM lunch_product product
-                RIGHT JOIN res_users users ON users.company_id = product.company_id -- multi company
-                LEFT JOIN lunch_product_favorite_user_rel fav ON fav.product_id = product.id AND fav.user_id = users.id
-                JOIN res_groups_users_rel groups ON groups.uid = users.id -- only generate for internal users
-                LEFT JOIN lunch_order l_order ON l_order.product_id = product.id AND l_order.user_id = users.id
+                INNER JOIN res_users users ON users.company_id = product.company_id -- multi company
+                INNER JOIN res_groups_users_rel groups ON groups.uid = users.id -- only generate for internal users
+                LEFT JOIN LATERAL (select max(date) AS last_order_date FROM lunch_order where user_id=users.id and product_id=product.id) AS orders ON TRUE
```

Comment:
Isn't this the same as a lateral cartesian product?

Reply:
As far as I know, this view is only used on a per-user basis, which makes the inner joins efficients.